### PR TITLE
feat(ES6 classes): adds the ES6 class syntax and functionality.

### PR DESCRIPTION
**Pull request by arjunkathuria**

### CODE REVIEW

**Comment by arjunkathuria** [comment]
@runspired merge ?

**Comment by runspired** [comment]
Merge but needs a follow up PR to address code comment.

File: src/recognizers/tap.js [hunk 1]
```diff
@@ -21,38 +20,39 @@ import getDistance from '../inputjs/get-distance';
...
-  defaults: {
-    event: 'tap',
-    pointers: 1,
-    taps: 1,
-    interval: 300, // max time between the multi-tap taps
-    time: 250, // max time of the pointer to be down (like finger on the screen)
-    threshold: 9, // a minimal movement is ok, but keep it low
-    posThreshold: 10 // a multi-tap can be a bit off the initial position
-  },
+  get defaults() {
```
Comment (runspired) [inline src/recognizers/tap.js:41]: this should stay as a prop, with ES6 classes you can set props from within the constructor method.

Comment (arjunkathuria) [inline src/recognizers/tap.js:41]: Okiee dokes 😉

**Pull request merged**